services/venice-controller/src/main/java/com/linkedin/venice/controller/server/StoresRoutes.java

```diff
@@ -327,6 +327,10 @@ public void internalHandle(Request request, StoreResponse veniceResponse) {
         if (storeInfo.getBackupVersionRetentionMs() < 0) {
           storeInfo.setBackupVersionRetentionMs(admin.getBackupVersionDefaultRetentionMs());
         }
+        // This is the only place the default value of maxRecordSizeBytes is set for StoreResponse for VPJ and Consumer
+        if (storeInfo.getMaxRecordSizeBytes() < 0) {
```

Comment:
Nit,
If this logic is being deployed later than VPJ, VPJ will receive `-1` for this field, right?
I wondered if StoreResponse doesn't contain this field at all because of the deploying order, what would VPJ receive? I guess it might receive `0` as `0` is the default value for `Integer` if the json response doesn't contain such field, so maybe we should update the default value of `StoreInfo#maxRecordSizeBytes` to be `VeniceWriter#UNLIMITED_MAX_RECORD_SIZE`?

Reply:
Nice catch. Fixed.